### DISCUSSION TITLE
Define a placeholder text for a RichTextView

### DISCUSSION THREAD

**Discussion by martindufort** [danielsaidi/RichTextKit#174] [Q&A]
What would be the best way to define a placeholder text for an empty RichTextView. 
And that placeholder would dissappear upon first char and appear on empty?

Thanks

**Comment by S1D1T1** [comment]
do you mean the light gray text such as “Username” that instructs the user what to enter?

![](https://docs-assets.developer.apple.com/published/db7d89f1d4ec1edf1cf57d9a07dca96b/SwiftUI-TextField-roundedBorderStyle@2x.png)


I implemented that with RichTextEditor. See my code here:  https://github.com/danielsaidi/RichTextKit/discussions/165

**Comment by danielsaidi** [comment]
@martindufort @S1D1T1 While the `NSTextView` approach is a great find, I'm not sure a corresponding API exists for `UITextView`. If it does, I think we should add a placeholder property to the component protocol. 👍

**Comment by S1D1T1** [comment]
agreed. UITextview is more of a challenge - and I do need it for my iOS release. Someone once posted a trick for that a long while back. We should get that guy's [help](https://danielsaidi.com/blog/2017/10/31/uitextview-placeholder).

**Comment by martindufort** [comment]
Ok so what is the best way to tackle this apart from setting a "special color" to the placeholder string. 
Can we insert special "placeholder" key attribute over the range of the placeholder and then look for it within the text?
